doc/man/task-sync.5.in

```diff
@@ -5,21 +5,97 @@ task-sync \- A discussion and tutorial for the various task(1) data
 synchronization capabilities.
 
 .SH INTRODUCTION
-Taskwarrior has several sync options, both external and built in.  If you wish
-to sync your data, choose one method only; mixing methods is going to lead to
-problems.  Each of the methods discussed have their own strengths.
+
+Taskwarrior can "synchronize" your tasks to a server. This has a few benefits:
+.br
+  - Makes your tasks accessible from multiple systems.
+.br
+  - Provides a backup of your tasks.
+.br
+  - Allows internal storage to be reused.
+
+.SH LOCAL SYNCHRONIZATION
+Taskwarrior can sync your tasks locally. While this setup does not make your
+tasks accessible from multiple systems, it does provide a convenient backup
+and allow internal storage to be reused. If you later decide to sync to a
```

Comment:
I've had a few cases with the existing Taskwarrior where I've messed up my local replica and just deleted it and re-sync'd from freecinc.com, so that's the sort of backup I'm thinking of. But, hopefully Taskchampion is more reliable than TDB2 has been!

The strongest argument for this is actually the storage-reuse argument: TC stores all operations that have occurred since the last sync, as well as the current set of tasks. So if you never run `task sync`, that list of tasks grows linearly. I'm up to about 9000 operations in my work instance over the last 2 months. The growth rate is probably pretty slow, so maybe this isn't a big deal. But that's what I mean by "allows internal storage to be reused": once a sync occurs, all of those operations can be discarded, so if a replica is regularly sync'd then its site is roughly proportional to the number of un-expired tasks, so O(1) and not O(n).

As-written, the "local" server will consume O(n) space as it never discards old versions, but this just needs implementation of such discarding to fix. The taskchampion-sync-server does implement such discarding.

So I suppose the options are:
 * Remove support for local sync's, and just accept the linear storage growth, with the solution being sync'ing.
 * Remove support for local sync's, and if sync is not enabled then drop all operations. This would mean that a replica can not be automatically migrated from "not synced" to "synced" without an export/import.
 * Keep support for local sync's, but "hide" it by doing so automatically, within the .task directory.

What do you think?

Reply:
I'm afraid I'm still confused. "Storage-reuse" means freeing disk space, right? It's not clear to me how local synchronization frees disk space if you're just transferring the operations from one path to another. Is the idea that your backup is on a larger disk? Or is this about tricking taskchampion into discarding operations?

Reply:
Yes, that's correct -- the server uses "snapshots' to capture the state of the taskdb at a point in time, and can then discard operations prior to that time. See https://github.com/GothenburgBitFactory/taskwarrior/blob/develop/taskchampion/docs/src/snapshots.md for details.

But you're right that "backups" is not a particularly strong argument for using local sync. What do you think of the options?

Reply:
Of the options you suggested, I think I'm most in favor of the third option:

> Keep support for local sync's, but "hide" it by doing so automatically, within the .task directory.

Does this mean that taskwarrior would automatically periodically create snapshots and discard operations with no user intervention? Seems like we could piggyback off the garbage collection process and use the same settings, expiration time, etc., right?